bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/wrappers/veneer/BulkReadVeneerApi.java

```diff
@@ -77,13 +79,22 @@ public Thread newThread(Runnable r) {
     this.callContext = callContext;
 
     this.batchers = new HashMap<>();
-    this.cleanupBarrier = new AtomicLong();
-    this.cleanupBarrier
-        .incrementAndGet(); // wait for sendOutstanding to signal before cleaning up the batcher map
+    this.cleanupBarrier = new AtomicLong(1);
+  }
+
+  @Override
+  public void close() {
+    // TODO: use closeAsync after https://github.com/googleapis/gax-java/pull/1423 is available
+    if (!isClosed.compareAndSet(false, true)) {
+      return;
+    }
+    notifyArrival();
   }
 
   @Override
   public ApiFuture<Result> add(ByteString rowKey, @Nullable Filters.Filter filter) {
+    Preconditions.checkState(!isClosed.get());
```

Comment:
Maybe also add the error message here? "can't add request when the bulk read is closed."

Reply:
done